src/AppInstallerCommonCore/Public/winget/ManifestValidation.h

```diff
@@ -71,59 +73,61 @@ namespace AppInstaller::Manifest
+        ValidationError(AppInstaller::StringResource::StringId message, std::string context, std::string_view value) :
+            Message(std::move(message)), Context(std::move(context)), Value(value) {}
 
-        ValidationError(std::string message, std::string field, std::string_view value) :
-            Message(std::move(message)), Field(std::move(field)), Value(value) {}
+        ValidationError(AppInstaller::StringResource::StringId message, std::string context, std::string value) :
+            Message(std::move(message)), Context(std::move(context)), Value(std::move(value)) {}
 
-        ValidationError(std::string message, std::string field, std::string value) :
-            Message(std::move(message)), Field(std::move(field)), Value(std::move(value)) {}
+        ValidationError(AppInstaller::StringResource::StringId message, std::string context, std::string value, Level level) :
+            Message(std::move(message)), Context(std::move(context)), Value(std::move(value)), ErrorLevel(level) {}
 
-        ValidationError(std::string message, std::string field, std::string value, Level level) :
-            Message(std::move(message)), Field(std::move(field)), Value(std::move(value)), ErrorLevel(level) {}
+        ValidationError(AppInstaller::StringResource::StringId message, std::string context, std::string value, size_t line, size_t column) :
+            Message(std::move(message)), Context(std::move(context)), Value(std::move(value)), Line(line), Column(column) {}
 
-        ValidationError(std::string message, std::string field, std::string value, size_t line, size_t column) :
-            Message(std::move(message)), Field(std::move(field)), Value(std::move(value)), Line(line), Column(column) {}
+        ValidationError(AppInstaller::StringResource::StringId message, std::string context, std::string value, size_t line, size_t column, Level level) :
+            Message(std::move(message)), Context(std::move(context)), Value(std::move(value)), Line(line), Column(column), ErrorLevel(level) {}
 
-        ValidationError(std::string message, std::string field, std::string value, size_t line, size_t column, Level level) :
-            Message(std::move(message)), Field(std::move(field)), Value(std::move(value)), Line(line), Column(column), ErrorLevel(level) {}
+        std::string GetErrorMessage() const;
 
-        static ValidationError MessageWithFile(std::string message, std::string file)
+        static ValidationError MessageWithFile(AppInstaller::StringResource::StringId message, std::string file)
         {
             ValidationError error{ message };
             error.FileName = file;
             return error;
         }
 
-        static ValidationError MessageFieldWithFile(std::string message, std::string field, std::string file)
+        static ValidationError MessageFieldWithFile(AppInstaller::StringResource::StringId message, std::string context, std::string file)

```

Comment:
>MessageField [](http://example.com/codeflow?start=31&length=12)

I guess this should be called MessageWithContextFile now?

And same for below
MessageWithContextValueFile